test/Sema/superfluously-public-imports.swift

```diff
@@ -0,0 +1,157 @@
+// RUN: %target-swift-frontend -emit-module %t/AliasesBase.swift -o %t
+// RUN: %target-swift-frontend -emit-module %t/Aliases.swift -o %t -I %t
+// RUN: %target-swift-frontend -emit-module %t/UnusedImport.swift -o %t -I %t
+// RUN: %target-swift-frontend -emit-module %t/UnusedPackageImport.swift -o %t -I %t
+// RUN: %target-swift-frontend -emit-module %t/ImportNotUseFromAPI.swift -o %t -I %t
+// RUN: %target-swift-frontend -emit-module %t/ImportUsedInPackage.swift -o %t -I %t
+
+/// Check diagnostics.
+// RUN: %target-swift-frontend -typecheck %t/Client.swift -I %t \
+// RUN:   -enable-experimental-feature AccessLevelOnImport -verify \
+// RUN:   -package-name pkg -Rmodule-api-import -swift-version 6
+// RUN: %target-swift-frontend -typecheck %t/Client_Swift5.swift -I %t \
+// RUN:   -enable-experimental-feature AccessLevelOnImport -verify \
+// RUN:   -swift-version 5
+
+//--- DepUsedFromInlinableCode.swift
+public struct TypeUsedFromInlinableCode {}
+public func funcUsedFromInlinableCode() {}
+
+public func funcUsedFromDefaultValue() -> Int { 42 }
+
+//--- DepUsedInSignature.swift
+public struct TypeUsedInSignature {}
+public protocol ComposedProtoA {}
+public protocol ComposedProtoB {}
+
+//--- Exportee.swift
+public struct ExportedType {}
+
+//--- Exporter.swift
+@_exported import Exportee
+
+//--- ConformanceBaseTypes.swift
+public protocol Proto {}
+public struct ConformingType {
+    public init () {}
+}
+
+//--- ConformanceDefinition.swift
+import ConformanceBaseTypes
+extension ConformingType : Proto  {}
+
+//--- AliasesBase.swift
+open class Clazz {}
+
+//--- Aliases.swift
+import AliasesBase
+public typealias ClazzAlias = Clazz
+
+//--- UnusedImport.swift
+
+//--- UnusedPackageImport.swift
+
+//--- ImportNotUseFromAPI.swift
+public struct NotAnAPIType {}
+public func notAnAPIFunc() -> NotAnAPIType { return NotAnAPIType() }
+
+//--- ImportUsedInPackage.swift
+public struct PackageType {}
+public func packageFunc() -> PackageType { return PackageType() }
+
+//--- Client_Swift5.swift
+/// No diagnostics should be raised on the implicit access level.
+import UnusedImport // expected-error {{ambiguous implicit access level for import of 'UnusedImport'; it is imported as 'public' elsewhere}}
+public import UnusedImport // expected-warning {{public import of 'UnusedImport' was not used in public declarations or inlinable code}} {{1-7=internal}}
+// expected-note @-1 {{imported 'public' here}}
+
+//--- Client.swift
+public import DepUsedFromInlinableCode
+public import DepUsedInSignature
+public import Exporter
+public import ConformanceBaseTypes
+public import ConformanceDefinition
+public import AliasesBase
+public import Aliases
+
+public import UnusedImport // expected-warning {{public import of 'UnusedImport' was not used in public declarations or inlinable code}} {{1-8=}}
+public import UnusedImport // expected-warning {{public import of 'UnusedImport' was not used in public declarations or inlinable code}} {{1-8=}}
```

Comment:
repeated twice L88 and L89

Reply:
The repetitions on this import are intentional. I wanted to make sure we reports all imports, no matter the modifiers.